framework/src/main/kotlin/com/cultureamp/eventsourcing/EventStore.kt

```diff
@@ -12,13 +12,15 @@ interface EventStore {
     // TODO this should be removed and implemented as separate threads/workers that poll the event-store
     fun notifyListeners(newEvents: List<Event>, aggregateId: UUID) {
         newEvents.forEach { event ->
-            listeners.flatMap {it.handlers.filterKeys { it.isInstance(event.domainEvent) }.values}.forEach { it(event.domainEvent, aggregateId) }
+            listeners.flatMap { it.handlers.filterKeys { it.isInstance(event.domainEvent) }.values }
+                .forEach { it(event.domainEvent, aggregateId) }
         }
     }
+
     fun setup()
 
     /**
-     * Replay all the events in the store on the project function
+     * Replay all the events in the store on the project function, for an aggregateType
      */
-    fun replay(project: (Event) -> Unit)
+    fun replay(aggregateType: String, project: (Event) -> Unit)
```

Comment:
I'm not sure if I understand why we need this. I can see reference to this in the other PR against the bookmarks project, so I'll check that out but theoretically this shouldn't be needed.

Reply:
Leaving this for now because we will need something like this for async projectors, but I don't think it's currently being used properly in the Bookmarks PR